### DISCUSSION TITLE
NiceGUI with Capacitor or Cordova

### DISCUSSION THREAD

**Discussion by bobwatcherx** [zauberzeug/nicegui#687] [Ideas / Feature Requests]
### Question

Is it possible to use the driver or package that is in the capacitor. And if Nicegui supports mobile then it will be 100 percent the same as Quasar JS. And if possible add documentation Also on nicegui . And how to set up like capacitor.config.json  in python . and how to set splashscreen too. And if possible to build android with capacitor with nicegui.😘That would be great thanks

https://quasar.dev/quasar-cli-webpack/developing-capacitor-apps/configuring-capacitor#capacitor-config-json

**Comment by falkoschindler** [comment]
So you're asking if NiceGUI can integrate with Capacitor to create native mobile applications with similar features and configurations to those built using Quasar JS.
Let me convert this question into a Feature Request. Maybe someone likes to try it out?

**Comment by rodja** [comment]
While NiceGUI is a powerful library for creating web-based user interfaces for Python applications, it is not specifically built for mobile app development or integration with Capacitor. Capacitor is designed for JavaScript/TypeScript-based web apps, while NiceGUI is a Python library. However, you can still create a responsive web app using NiceGUI that works well on mobile devices by adapting the layout, components, and styling for smaller screens.

To achieve a more native-like experience, you could use a WebView-based wrapper (such as WebViewGold for Android or WKWebView for iOS) to display your NiceGUI app inside a native app shell. This approach would allow you to create a mobile app that can be published to app stores, but it won't be as integrated as a Capacitor or React Native app.

As for splash screens, you would need to handle this on the wrapper side, as NiceGUI doesn't provide built-in support for splash screens. For example, if you're using WebViewGold, you can configure a splash screen by following their documentation.

**Comment by bobwatcherx** [comment]
if it works . I just discovered how capacitor can work for python. that would be great . and how to also display admob ads.Access the contact list. SMS . And push notifications. and all the features on Capacitor can run on Nicegui

**Reply by rodja** [reply]
Could you share the details on how Capacitor could work with Python?

**Reply by gykung** [reply]
May I ask if there's follow up on this? I'm interested in nicegui as a iOS app with native background location.
